pytest.ini

```diff
@@ -1,2 +1,6 @@
 [pytest]
 addopts = --color=yes --durations=5 --showlocals --verbose
+# Turn warnings for deprecated numpy type aliases into errors
+filterwarnings =
+  error:^`np\.[a-z]+` is a deprecated alias for the builtin.+:DeprecationWarning:mlflow
+  error:^`np\.[a-z]+` is a deprecated alias for the builtin.+:DeprecationWarning:tests
```

Comment:
This works as a test that ensures we don't use deprecated types aliases.

Reply:
We recently dropped support for python 3.6 and started using python 3.7 in CI, which allows us to use NumPy >=1.20.0 that raises warnings for deprecated type aliases.